src/huggingface_hub/file_download.py

```diff
@@ -1377,3 +1402,71 @@ def try_to_load_from_cache(
 
     cached_file = os.path.join(repo_cache, "snapshots", revision, filename)
     return cached_file if os.path.isfile(cached_file) else None
+
+
+def get_hf_file_metadata(
+    url: str,
+    use_auth_token: Union[bool, str, None] = None,
+    proxies: Optional[Dict] = None,
+    timeout: Optional[float] = 10,
+) -> HfFileMetadata:
+    """Fetch metadata of a file versioned on the Hub for a given url.
+
+    Args:
+        url (`str`):
+            File url, for example returned by [`hf_hub_url`].
+        use_auth_token (`str`, `bool`, *optional*):
+            A token to be used for the download.
+                - If `True`, the token is read from the HuggingFace config
+                  folder.
+                - If `False` or `None`, no token is provided.
+                - If a string, it's used as the authentication token.
+        proxies (`dict`, *optional*):
+            Dictionary mapping protocol to the URL of the proxy passed to
+            `requests.request`.
+        etag_timeout (`float`, *optional*, defaults to `10`):
```

Comment:
```suggestion
        etag_timeout (`float`, *optional*, defaults to 10):
```
Not sure what the convention is for hugginface_hub, but in Transformers we don't put numbers in code blocks, only objects.